rest-api-spec/src/main/resources/rest-api-spec/test/index/90_flat_object.yml

```diff
@@ -0,0 +1,234 @@
+---
```

Comment:
Please add tests for (as per description):
 - `simple_query_string` query 
 - `query_string` query 
 - `terms` query 
 - `terms_set` query
 - `multi_match` query

We could also add tests to make sure the unsupported queries are also properly failing, like wildcard query for example